### PR TITLE
AB#3600 -- Updated routes for adult-child apply-yourself page

### PR DESCRIPTION
**Pull request by AdamP-HRSDC-RHDCC**
### Description
Updated routes for adult-child apply-yourself page (Issue with /adult-child/apply-yourself  - Redirect to proper screen)

### Related Azure Boards Work Items
[AB#3600](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3600)

### Screenshots (if applicable)
n/a

### Checklist
- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e`

### Test Instructions
Confirm figma flow on DTC page with all children over/under 18 options selected.

### Additional Notes
n/a

### CODE REVIEW

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/apply-yourself.tsx [hunk 1]
```diff
@@ -44,6 +44,11 @@ export async function action({ context: { session }, params, request }: ActionFu
   const formData = await request.formData();
   const expectedCsrfToken = String(session.get('csrfToken'));
   const submittedCsrfToken = String(formData.get('_csrf'));
+  const state = loadApplyAdultChildState({ params, request, session });
+
+  if (state.dateOfBirth === undefined) {
+    throw new Error(`Invalid date of birth "${state.id}" state!`);
+  }
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/apply-yourself.tsx:51]:
```suggestion
  invariant(state.dateOfBirth, 'Expected state.dateOfBirth to be defined');
```

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

**Review by sebastien-comeau — DISMISSED** [review]

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/disability-tax-credit.tsx [hunk 2]
```diff
@@ -96,11 +96,15 @@ export async function action({ context: { session }, params, request }: ActionFu
...
+  if (parsedDataResult.data === DisabilityTaxCreditOption.Yes && state.allChildrenUnder18) {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/applicant-information', params));
   }
 
-  return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/applicant-information', params));
+  if (parsedDataResult.data === DisabilityTaxCreditOption.Yes) {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/apply-yourself', params));
+  }
+
+  return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/parent-or-guardian', params));
```
Comment (sebastien-comeau) [inline frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/disability-tax-credit.tsx:107]: Change the conditions.

```
IF "NO" THEN redirect "adult-child/parent-or-guardian" END

// at this point it's "YES"

IF NOT "allChildrenUnder18" THEN redirect "adult-child/apply-yourself" END

redirect "adult-child/applicant-information"
```

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/apply-yourself.tsx [hunk 3]
```diff
@@ -59,7 +63,17 @@ export async function action({ context: { session }, params, request }: ActionFu
     },
   });
 
-  return redirect(getPathById('$lang+/_public+/apply+/$id+/adult/applicant-information', params));
+  const ageCategory = getAgeCategoryFromDateString(state.dateOfBirth);
+
+  if (ageCategory === 'children') {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/contact-apply-child', params));
+  } else if (ageCategory === 'youth') {
+    return redirect(getPathById('$lang+/_public+/apply+/$id+/adult-child/living-independently', params));
```
[the remark this answers is not in the document]
Comment (AdamP-HRSDC-RHDCC) [reply frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/apply-yourself.tsx]: Changed "youth" to redirect to adult flow

File: frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/disability-tax-credit.tsx (hunk 2 above)
Comment (AdamP-HRSDC-RHDCC) [reply frontend/app/routes/$lang+/_public+/apply+/$id+/adult-child/disability-tax-credit.tsx]: Updated conditionals

**Review by sebastien-comeau — APPROVED** [review]

**Pull request merged**